sis-multi.el

```diff
@@ -111,138 +81,11 @@ FN is invoked.")
-    ;; avoid override user customized sis-do-set
-    (unless (functionp sis-do-set)
-      (setq sis-do-set (sis--mk-set-fn))))
-
-  ;; successfully inited
-  (when (and (functionp sis-do-get)
-             (functionp sis-do-set))
-    ;; a t `sis--ism' means customized by `sis-do-get' and `sis-do-set'
-    (unless sis--ism (setq sis--ism t)))
-
-  ;; just inited, successfully or not
-  (setq sis--ism-inited t))
-
-(defmacro sis--ensure-ism (&rest body)
-  "Only run BODY with valid ism."
-  `(progn
-     (unless sis--ism-inited
-       (sis--init-ism))
-     (when sis--ism
-       ,@body)))
-
-(defmacro sis--ensure-dir (&rest body)
-  "Ensure BODY run in home directory."
-  `(let ((default-directory "~"))
-     ,@body))
-
-(defun sis--mk-get-fn ()
-  "Make a function to be bound to `sis-do-get'."
-  (cond
-   (; EMP
-    (equal sis--ism 'emp)
-    #'mac-input-source)
-   (; external ism
-    sis--ism
-    (lambda ()
-      (sis--ensure-dir
-       (string-trim (shell-command-to-string sis--ism)))))))
-
-(defun sis--mk-set-fn ()
-  "Make a function to be bound to `sis-do-set'."
-  (cond
-   (; EMP
-    (equal sis--ism 'emp)
-    (lambda (source) (mac-select-input-source source)))
-   (; external ism
-    sis--ism
-    (lambda (source)
-      (sis--ensure-dir
-       (start-process "set-input-source" nil sis--ism source))))))
-
-(defun sis--update-state (source)
-  "Update input source state.
-
-SOURCE should be 'english or 'other."
-
-  (setq sis--previous sis--current)
-  (setq sis--current source)
-  (when (not (eq sis--previous sis--current))
-    (run-hooks 'sis-change-hook)))
-
-(defsubst sis--normalize-to-lang (lang)
-  "Normalize LANG in the form of source id or lang to lang."
-  (cond
-   (; english
-    (member lang (list 'english sis-english-source))
-    'english)
-   (; hangul
-    (member lang (list 'hangul sis-hangul-source))
-    'hangul)
-   (; japanese
-    (member lang (list 'japanese sis-japanese-source))
-    'japanese)))
-
-(defsubst sis--normalize-to-source (source)
-  "Normalize SOURCE in the form of source id or lang to source."
-  (cond
-   (; english
-    (member source (list 'english sis-english-source))
-    sis-english-source)
-   (; hangul
-    (member source (list 'hangul sis-hangul-source))
-    sis-hangul-source)
-   (; japanese
-    (member source (list 'japanese sis-japanese-source))
-    sis-japanese-source)))
-
-(defsubst sis--get ()
-  "Get the input source id."
-  (sis--ensure-ism
-   (sis--update-state (sis--normalize-to-lang (funcall sis-do-get)))))
-
 (defsubst sis--set (source)
   "Set the input source according to source SOURCE."
-  (sis--ensure-ism
-   (sis--update-state (sis--normalize-to-lang source))
-   (funcall sis-do-set (sis--normalize-to-source source))
-   ))
-
-(defun sis--set-english ()
-  "Function to set input source to `english'."
-  (sis--set 'english))
+  (if (eq source 'en)
+      (deactivate-input-method)
+    (activate-input-method source)))
```

Comment:
Can't we get rid of this treatment? It deteriorates overall consistency, but it may be worth to notice that English is the default one in Emacs also. So the special treatment would be justifiable.